src/include/ZividPython/DataModelWrapper.h

```diff
@@ -16,77 +26,233 @@ namespace ZividPython
             return s.template get<Target>();
         }
 
+        template<typename T>
+        bool hasValue(const T &leaf)
+        {
+            if constexpr(Zivid::DataModel::IsOptional<T>::value)
+            {
+                return leaf.hasValue();
+            }
+            return true;
+        }
+
+        template<typename T>
+        struct TypeName
+        {
+            static constexpr const char *value{ "(collections.abc.Iterable,)" };
+        };
```

Comment:
Not sure I fully understand the issue. Would you like me to add a TODO-comment here? If so, could you write it out for me?
(I think it's cleaner if I'm the only one pushing fixups to this branch)

Reply:
This part of the code determines the types we should use when generating python code. So a `std::String` will become `str`, etc. Since this particular case is the default cause, we should make sure that this will throw if there is no special case.
I removed the throw here becuase I did not want to get stuck at this when developing, but we should make one more attempt at adding a special case that handles containers. If someone adds a new type in the future the runtime type check in python should not get a check against `collections.abc.Iterable`, but agains `SomeNewType`

Reply:
Can you have a shot at doing this, on a separate branch? If successful I can cherry-pick it into this branch.

Reply:
The problem is that I do not really know what the type is in C++. I am not sure how to print the type we evaluate either.

Reply:
Alright, I can try to spend some time understanding the DataModelWrapper, and then see what I can do.

Reply:
Which is good because it led me to catch this bug: 
![image](https://user-images.githubusercontent.com/50012606/96145517-94c31c80-0f05-11eb-8c40-d2b3a6779b22.png)

It always worked to set max_capture_time through the constructor of SuggestSettingsParameter, but changing the value via the property was broken because it mistakenly had the type "collections.abc.Iterable" instead of "timedelta".

Reply:
Nice catch

Reply:
Of course, the solution does not work on Windows... ;)

Reply:
Got it working on windows. Pushed the solution now.